tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -2431,6 +2431,124 @@ def func(self, xp, a):
+        'is_module': [True, False],
+    })
+    # Special values
+    + chainer.testing.product({
+        'in_shapes': [((2, 3), (2, 3))],
+        'in_dtypes,out_dtype': (
+            _make_same_in_out_dtypes(2, chainerx.testing.float_dtypes)),
+        'input_lhs': ['random', float('inf'), -float('inf'), float('nan')],
+        'input_rhs': ['random', float('inf'), -float('inf'), float('nan')],
+        'is_module': [False],
+        'skip_backward_test': [True],
+        'skip_double_backward_test': [True],
+    })
+))
+class TestPow(BinaryMathTestBase, op_utils.NumpyOpTest):
+
+    def setup(self):
+        super().setup()
+        dtype1, dtype2 = self.in_dtypes
+
+        self.check_forward_options.update({'rtol': 5e-3, 'atol': 5e-3})
+        self.check_backward_options.update({'rtol': 5e-3, 'atol': 5e-3})
+        self.check_double_backward_options.update(
+            {'rtol': 5e-3, 'atol': 5e-3})
+
+    def func(self, xp, a, b):
+        if self.is_module:
+            return xp.power(a, b)
+        else:
+            return a ** b
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'in_dtypes,scalar_type,out_dtype':
+            _in_out_dtypes_float_arithmetic_scalar,
+        'input': ['random'],
+        'scalar_value': [2.0, 3.0],
+        'is_module': [False],
+        'is_scalar_rhs': [True, False],
+    })
+    # Type combinations
+    + chainer.testing.product({
+        'shape': [(2, 3)],
+        'in_dtypes,scalar_type,out_dtype':
+            _in_out_dtypes_float_arithmetic_scalar,
+        'input': ['random'],
+        'scalar_value': [2.0, 3.0],
+        'is_module': [False],
+        'is_scalar_rhs': [True, False],
+    })
+    # is_module
+    + chainer.testing.product({
+        'shape': [(2, 3)],
+        'in_dtypes,scalar_type,out_dtype':
+            _in_out_dtypes_float_arithmetic_scalar,
+        'input': ['random'],
+        'scalar_value': [1.0],
+        'is_module': [True, False],
+        'is_scalar_rhs': [True, False],
+    })
+    # Special values
+    + chainer.testing.product({
+        'shape': [(2, 3)],
+        'in_dtypes,scalar_type,out_dtype':
+            _in_out_dtypes_float_arithmetic_scalar,
+        'input': [float('inf')],
+        'scalar_value': [1, 2],
+        'is_module': [False],
+        'is_scalar_rhs': [False],
+        'skip_backward_test': [True],
+        'skip_double_backward_test': [True],
+    })
+))
+class TestPowScalar(MathScalarTestBase, op_utils.NumpyOpTest):
+
+    def func_scalar(self, xp, a, scalar):
+        if self.is_module:
+            if self.is_scalar_rhs:
+                return xp.power(a, scalar)
+            else:
+                return xp.power(scalar, a)
+        else:
+            if self.is_scalar_rhs:
+                return a ** scalar
+            else:
+                return scalar ** a
```

Comment:
Shouldn't we test non-floats such as integers too? Note that an int raised to a negative int will invoke a `ValueError` for NumPy while for CuPy it won't. I think it's okay for ChainerX to follow CuPy.

Reply:
Fixed.